cibuildwheel/__main__.py

```diff
@@ -73,6 +72,8 @@ def main():
                 platform = 'macos'
             elif sys.platform == 'win32':
                 platform = 'windows'
+            else:
+                platform = None
```

Comment:
the assignment shall be moved to the same level as the `ci` assignment to fix all cases. This only fixes the case where we're running in CI but `sys.platform` returns an unsupported value.
If we wan't a different message for each case, could be modified as follow:
```
        ci = strtobool(os.environ.get('CI', 'false')) or 'BITRISE_BUILD_NUMBER' in os.environ or 'AZURE_HTTP_USER_AGENT' in os.environ
        if not ci:
            print('cibuildwheel: Unable to detect platform. cibuildwheel should run on your CI server, '
                  .........,
                  file=sys.stderr)
            exit(2)
        if sys.platform.startswith('linux'):
            platform = 'linux'
        elif sys.platform == 'darwin':
            platform = 'macos'
        elif sys.platform == 'win32':
            platform = 'windows'
        else:
            print('cibuildwheel: Unable to detect platform from "sys.platform"'
                  .........,
                  file=sys.stderr)
            exit(2)

```

Reply:
I change it, but in another way. Your proposition create code repetition.

Reply:
@Czaki, in fact you're using my exact first proposition. The second proposition doesn't duplicate code.
Full picture:
```

    detect_obsolete_options()

    if args.platform != 'auto':
        platform = args.platform
    else:
        ci = strtobool(os.environ.get('CI', 'false')) or 'BITRISE_BUILD_NUMBER' in os.environ or 'AZURE_HTTP_USER_AGENT' in os.environ
        if not ci:
            print('cibuildwheel: Unable to detect platform. cibuildwheel should run on your CI server, '
                  'Travis CI, AppVeyor, Azure Pipelines and CircleCI are supported. You can run on your '
                  'development machine or other CI providers using the --platform argument. Check --help '
                  'output for more information.',
                  file=sys.stderr)
            exit(2)
        if sys.platform.startswith('linux'):
            platform = 'linux'
        elif sys.platform == 'darwin':
            platform = 'macos'
        elif sys.platform == 'win32':
            platform = 'windows'
        else:
            print('cibuildwheel: Unable to detect platform from "sys.platform" in a CI environment. You can run '
                  'cibuildwheel using the --platform argument. Check --help output for more information.',
                  file=sys.stderr)
            exit(2)

    output_dir = args.output_dir
```

Reply:
Ok, I see the difference and agree that it is better.